build.gradle.kts

```diff
@@ -0,0 +1,375 @@
+                }
+            }
+
+            targetTestTask.dependsOn(taskName)
+        }
+    }
+
+    // Include resources from JS and Metadata (common) into the JS JAR
+    for (target in listOf(kotlin.targets.js, kotlin.targets.metadata)) {
+        for (sourceSet in target.compilations.main.kotlinSourceSets) {
+            for (it in sourceSet.resources.srcDirs) {
+                tasks.getByName<Jar>("jsJar").from(it)
+            }
+        }
+    }
+
+    // Only run JS tests if not in windows
+    if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+        tasks.getByName("jsTest").dependsOn("runMocha")
+    }
+
+    group = "com.soywiz"
+    version = properties["projectVersion"].toString()
+
+    // Publishing
+    val publishUser = (rootProject.findProperty("BINTRAY_USER") ?: project.findProperty("bintrayUser") ?: System.getenv("BINTRAY_USER"))?.toString()
+    val publishPassword = (rootProject.findProperty("BINTRAY_KEY") ?: project.findProperty("bintrayApiKey") ?: System.getenv("BINTRAY_API_KEY"))?.toString()
+
+    apply(plugin = "maven-publish")
+
+    if (publishUser != null && publishPassword != null) {
+        extensions.getByType<PublishingExtension>().apply {
+            repositories {
+                maven {
+                    credentials {
+                        username = publishUser
+                        setPassword(publishPassword)
+                    }
+                    url = uri("https://api.bintray.com/maven/soywiz/soywiz/${project.property("project.package")}/")
+                }
+            }
+            afterEvaluate {
+                configure(publications) {
+                    this as MavenPublication
+                    pom.withXml {
+                        this.asNode().apply {
+                            appendNode("name", project.name)
+                            appendNode("description", project.property("project.description"))
+                            appendNode("url", project.property("project.scm.url"))
+                            appendNode("licenses").apply {
+                                appendNode("license").apply {
+                                    appendNode("name").setValue(project.property("project.license.name"))
+                                    appendNode("url").setValue(project.property("project.license.url"))
+                                }
+                            }
+                            appendNode("scm").apply {
+                                appendNode("url").setValue(project.property("project.scm.url"))
+                            }
+
+                            // Changes runtime -> compile in Android's AAR publications
+                            if (pom.packaging == "aar") {
+                                val nodes = this.getAt(QName("dependencies")).getAt("dependency").getAt("scope")
+                                for (node in nodes) {
+                                    (node as Node).setValue("compile")
```

Comment:
Not in this project, but in other projects with dependencies, I have noticed that the pom generated from android for aar packaging all the dependencies are "scope". I have added this to preserve transitive dependencies when including the `-android` artifact in non-common projects. Someone experienced with Android have a better solution here?